### PR TITLE
[Auto Parallel] Inter api publication and add docs.

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR Category
Auto Parallel

### PR Types
Docs

### Description
Pcard-73145
Inter api publication and add docs.

### CODE REVIEW

**Review by deepllz — APPROVED** [review]
LGTM

**Review by jeff41404 — DISMISSED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/intermediate/parallelize.py [hunk 1]
```diff
@@ -13,16 +13,228 @@
...
+                a layer name to a `SplitPoint`, note that the layer name could be written in regular format. The
+                pipeline parallel will exactly split the model at the point indicated by the map. If the `split_spec`
+                is a string, it contains the prefix of a set of layers. The pipeline parallel will automatically split
+                the model evenly at target layer. The `global_spec` is a string indicating a layer that contains global
+                tensors, which will be duplicated through all stages of the pipeline parallel. Some valid pp_config
+                can be list these:  {"split_spec": "llama.layers", "global_spec": "llama.global_layer"}
+                or {"split_spec": {"llama.layers.1": SplitPoint.END}}.
+
+            Note: if the mesh is `None` or neither of `dp_config`, `mp_config` and `pp_config` is in the config, this
+            api will do nothing but return the model and optimizer passed in.
```
Comment (sunzhongkai588) [inline python/paddle/distributed/auto_parallel/intermediate/parallelize.py]:
```suggestion
    Note:
        if the mesh is `None` or neither of `dp_config`, `mp_config` and `pp_config` is in the config, this 
        api will do nothing but return the model and optimizer passed in.
```

**Review by sunzhongkai588 — COMMENTED** [review]
还得为 API 标注类型提示，参考 [Python 类型提示标注规范](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/style_guide_and_references/type_annotations_specification_cn.html)

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM for docs

File: python/paddle/distributed/auto_parallel/intermediate/parallelize.py [hunk 2]
```diff
@@ -11,18 +11,242 @@
...
+
 
 def parallelize(
-    model, optimizer, mesh=None, dp_config=None, mp_config=None, pp_config=None
-):
-    # TODO(yaliu): global mesh and split axis support
+    model: paddle.nn.Layer,
+    optimizer: paddle.optimizer.Optimizer | None = None,
+    mesh: paddle.distributed.ProcessMesh | None = None,
+    config: dict | None = None,
```
Comment (SigureMo) [inline python/paddle/distributed/auto_parallel/intermediate/parallelize.py]: config 内参数具体是什么类型的？泛型需要标明具体参数类型，是 `dict[str, str]` 还是？

File: python/paddle/distributed/auto_parallel/intermediate/parallelize.py [hunk 3]
```diff
@@ -11,18 +11,242 @@
...
+) -> (paddle.nn.Layer, paddle.optimizer.Optimizer):
```
Comment (SigureMo) [inline python/paddle/distributed/auto_parallel/intermediate/parallelize.py]: 返回的是 tuple？那要写成 `tuple[paddle.nn.Layer, paddle.optimizer.Optimizer]`

File: python/paddle/distributed/auto_parallel/intermediate/parallelize.py (hunk 2 above)
Comment (FeixLiu) [reply python/paddle/distributed/auto_parallel/intermediate/parallelize.py]: done

File: python/paddle/distributed/auto_parallel/intermediate/parallelize.py (hunk 3 above)
Comment (FeixLiu) [reply python/paddle/distributed/auto_parallel/intermediate/parallelize.py]: done

**Review by SigureMo — DISMISSED** [review]
LGTMeow

**Review by SigureMo — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]

**Pull request merged**